### PR TITLE
Moved the existing builders from the tools/builders to the workspace.

### LINKED ISSUE #414: Restructure Workspace builders
# Feature Request

## Summary

Currently, we cannot share common logic between builders and other libraries. So we had to duplicate this logic.

## Feature Description

Currently, we cannot share logic among other libraries and parts of our monorepo. The `@dynatrace/barista-builders` are getting built in a post-install step with a `tsc` typescript compile. The `outDir` is set to `node_modules/@dynatrace/barista-builders` so that our angular cli can use the compiled files directly without publishing it to npm. 

The problem with `tsc` is that it does not bundle files. So we need a module bundling to add files that are not located under the package root dir with typescript path aliases.

The current solution was more built to be a starting point that needs further improvements.

### PR DESCRIPTION
**Pull request by lukasholzer**
Now it is possible to share logic between libraries and the builders. Because we use for the workspace a module bundler (webpack), where we can specify multiple entry points. All workspace dependencies that are not an entry file are getting bundled into them. This is a part of issue #621.

Fixes #414

### <strong>Pull Request</strong>

<hr>
Hi, thank you for contributing to Barista with this pull request (PR).

To ensure a fast process and merging of your PR please make sure it fulfills the
coding standards and contribution guidelines.

- A feature proposal has been provided, discussed and approved first.
- There is a meaningful description of the issue in GitHub (Screenshots are
  often helpful).
- If the PR introduces breaking-changes or deprecations it matches the following
  guidelines.
  - The commit message follows our commit guidelines.
  - Tests for the changes have been added (for bug fixes / features).
  - Docs have been added / updated (for bug fixes / features).

Please choose the type appropriate for the changes below: <br>

#### Checklist

- [ ] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by lukasholzer** [comment]
blocked until this issue #851 is merged because paths depend on each other. Might need adaption then.

Furthermore double check the `entrypoint.sh` for the barista preview.

File: libs/tools/shared/src/interfaces/package-json.ts [hunk 1]
```diff
@@ -16,10 +16,15 @@
 
 export interface PackageJson {
   version?: string;
-  peerDependencies?: {
+  peerDependencies: {
```
Comment (ffriedl89) [inline libs/tools/shared/src/interfaces/package-json.ts:19]: not all packageJson files have a peerDependency field
same for dev or normal dependencies - those fields should be optional

Comment (lukasholzer) [reply libs/tools/shared/src/interfaces/package-json.ts:19]: Good point just copied it.

**Comment by lukasholzer** [comment]
> @dynatrace-oss/barista, the auto cherry picker failed. Please have a look at the failure and cherry pick by hand.

Successfully resolved merge conflict by cherry picking by hand.

**Pull request merged**